R/summary_funs.R

```diff
@@ -202,198 +209,223 @@ weighted_summary_means <- function(y_wobs_test, family, wdraws, mu, dis, cl_ref,
+  ##   if (stat %in% c("elpd", "mlpd", "gmpd")) {
+  ##     lppd[is.na(lppd_baseline)] <- NA
+  ##     n_notna.bs <- sum(!is.na(lppd_baseline))
+  ##   }
+  ##   n_notna <- sum(!is.na(lppd))
+  ##   n <- length(lppd)
+  ## } else {
+  ##   hasNA_y <- is.na(y_wobs_test$y_prop %||% y_wobs_test$y)
+  ##   if (!is.null(mu_baseline)) {
+  ##     # Compute the performance statistics using only those observations for
+  ##     # which both `mu` and `mu_baseline` are not `NA`:
+  ##     mu[is.na(mu_baseline)] <- NA
+  ##     mu_baseline[is.na(mu)] <- NA
+  ##     n_notna.bs <- sum(!is.na(mu_baseline) & !hasNA_y)
+  ##   }
+  ##   n_notna <- sum(!is.na(mu) & !hasNA_y)
+  ##   n <- length(mu)
+  ## }
+  ## if (!is.null(n_notna.bs) && getOption("projpred.additional_checks", FALSE)) {
+  ##   stopifnot(n_notna == n_notna.bs)
+  ## }
+  if (n_loo == 0) {
     return(list(value = NA, se = NA, lq = NA, uq = NA))
   }
-
-  if (is.null(wcv)) {
-    ## set default CV fold weights if not given
-    wcv <- rep(1, n)
-  }
-  ## ensure the CV fold weights sum to n_notna
-  wcv <- n_notna * wcv / sum(wcv)
-
   alpha_half <- alpha / 2
   one_minus_alpha_half <- 1 - alpha_half
 
   if (stat %in% c("elpd", "mlpd", "gmpd")) {
-    if (!is.null(lppd.bs)) {
-      value <- sum((lppd - lppd.bs) * wcv, na.rm = TRUE)
-      value.se <- weighted.sd(lppd - lppd.bs, wcv, na.rm = TRUE) *
-        sqrt(n_notna)
+    if (is.null(summaries_baseline)) {
+      lppd_baseline = 0
+    } else {
+      lppd_baseline = summaries_baseline$lppd
+    }
+    if (!is.null(summaries_fast) && sum(n_loo<n)) {
+      # subsampling difference estimator (Magnusson et al., 2020)
+      srs_diffe <- .srs_diff_est_w(y_approx = summaries_fast$lppd-lppd_baseline,
+                                   y = (lppd-lppd_baseline)[loo_inds],
+                                   y_idx = loo_inds)
+      value <- srs_diffe$y_hat
+      # combine estimates of var(y_hat) and var(y)
+      value_se <- sqrt(srs_diffe$v_y_hat + srs_diffe$hat_v_y)
     } else {
-      value <- sum(lppd * wcv, na.rm = TRUE)
-      value.se <- weighted.sd(lppd, wcv, na.rm = TRUE) *
-        sqrt(n_notna)
+      # full LOO estimator
+      value <- sum((lppd - lppd_baseline), na.rm = TRUE)
+      value_se <-sd(lppd - lppd_baseline, na.rm = TRUE) * sqrt(n)
     }
     if (stat %in% c("mlpd", "gmpd")) {
-      value <- value / n_notna
-      value.se <- value.se / n_notna
+      value <- value / n
+      value_se <- value_se / n
       if (stat == "gmpd") {
         value_gmpd <- exp(value)
-        # Delta method:
-        value_gmpd.se <- value.se * value_gmpd
+        # delta method
+        value_gmpd_se <- value_se * value_gmpd
       }
     }
-  } else if (stat %in% c("mse", "rmse")) {
+  } else if (stat %in% c("mse", "rmse", "R2")) {
     y <- y_wobs_test$y_prop %||% y_wobs_test$y
-    if (!all(y_wobs_test$wobs == 1)) {
-      wcv <- wcv * y_wobs_test$wobs
-      wcv <- n_notna * wcv / sum(wcv)
+    wcv <- y_wobs_test$wobs
```

Comment:
Given that the meaning of `wcv` changes with this PR, should we rename `wcv` to `wobs`? However, I haven't checked yet if that is possible (it could be that this results in naming conflicts with other objects called `wobs`).

Reply:
Within `summary_funs.R`, `wcv` occurs only in `get_stat()`. Outside of `summary_funs.R`, it occurs only in `.auc()` (where it would also make sense to rename this).

Reply:
renamed

Reply:
Thanks. I renamed in `.auc()` correspondingly.